test/e2e/run_security_labels.go

```diff
@@ -0,0 +1,325 @@
+	for _, option := range options {
+		option(&p)
+	}
+	if len(p.Ctrs) == 0 {
+		p.Ctrs = []*Ctr{getCtr()}
+	}
+	return &p
+}
+
+type podOption func(*Pod)
+
+func withHostname(h string) podOption {
+	return func(pod *Pod) {
+		pod.Hostname = h
+	}
+}
+
+func withCtr(c *Ctr) podOption {
+	return func(pod *Pod) {
+		pod.Ctrs = append(pod.Ctrs, c)
+	}
+}
+
+func withAnnotation(k, v string) podOption {
+	return func(pod *Pod) {
+		pod.Annotations[k] = v
+	}
+}
+
+// Ctr describes the options a kube yaml can be configured at container level
+type Ctr struct {
+	Name            string
+	Image           string
+	Cmd             []string
+	SecurityContext bool
+	Caps            bool
+	CapAdd          []string
+	CapDrop         []string
+	PullPolicy      string
+}
+
+// getCtr takes a list of ctrOptions and returns a Ctr with sane defaults
+// and the configured options
+func getCtr(options ...ctrOption) *Ctr {
+	c := Ctr{defaultCtrName, defaultCtrImage, defaultCtrCmd, true, false, nil, nil, ""}
+	for _, option := range options {
+		option(&c)
+	}
+	return &c
+}
+
+type ctrOption func(*Ctr)
+
+func withCmd(cmd []string) ctrOption {
+	return func(c *Ctr) {
+		c.Cmd = cmd
+	}
+}
+
+func withImage(img string) ctrOption {
+	return func(c *Ctr) {
+		c.Image = img
+	}
+}
+
+func withSecurityContext(sc bool) ctrOption {
+	return func(c *Ctr) {
+		c.SecurityContext = sc
+	}
+}
+
+func withCapAdd(caps []string) ctrOption {
+	return func(c *Ctr) {
+		c.CapAdd = caps
+		c.Caps = true
+	}
+}
+
+func withCapDrop(caps []string) ctrOption {
+	return func(c *Ctr) {
+		c.CapDrop = caps
+		c.Caps = true
+	}
+}
+
+func withPullPolicy(policy string) ctrOption {
+	return func(c *Ctr) {
+		c.PullPolicy = policy
+	}
+}
+
+var _ = Describe("Podman generate kube", func() {
+	var (
+		tempdir    string
+		err        error
+		podmanTest *PodmanTestIntegration
+	)
+
+	BeforeEach(func() {
+		tempdir, err = CreateTempDirInTempDir()
+		if err != nil {
+			os.Exit(1)
+		}
+		podmanTest = PodmanTestCreate(tempdir)
+		podmanTest.Setup()
+		podmanTest.SeedImages()
+
+	})
+
+	AfterEach(func() {
+		podmanTest.Cleanup()
+		f := CurrentGinkgoTestDescription()
+		processTestResult(f)
+	})
+
+	It("podman security labels", func() {
+		test1 := podmanTest.Podman([]string{"create", "--label", "io.containers.capabilities=setuid,setgid", "--name", "test1", "alpine", "echo", "test1"})
+		test1.WaitWithDefaultTimeout()
+		Expect(test1.ExitCode()).To(BeZero())
+
+		inspect := podmanTest.Podman([]string{"inspect", "test1"})
+		inspect.WaitWithDefaultTimeout()
+		Expect(inspect.ExitCode()).To(Equal(0))
+
+		ctr := inspect.InspectContainerToJSON()
+		caps := strings.Join(ctr[0].EffectiveCaps, ",")
+		Expect(caps).To(Equal("CAP_SETUID,CAP_SETGID"))
+	})
+
+	It("podman bad security labels", func() {
+		test1 := podmanTest.Podman([]string{"create", "--label", "io.containers.capabilities=sys_admin", "--name", "test1", "alpine", "echo", "test1"})
```

Comment:
Looks like we're not testing with an image, and I believe we should.

Reply:
I commit an image and test off the image, but if you are talking about an image built with a Containerfile, I could do that as well.

Reply:
Didn't notice the commit below. Thanks! LGTM

Reply:
Also added a Dockerfile build